packages/composite-checkout-wpcom/src/components/wp-contact-form.js

```diff
@@ -531,8 +368,64 @@ function joinNonEmptyValues( joinString, ...values ) {
 	return values.filter( value => value.length > 0 ).join( joinString );
 }
 
-const StateSelectWrapper = styled.div`
-	& select {
-		width: 100%;
+function contactDetailsFormat( { isDomainFieldsVisible } ) {
+	if ( isDomainFieldsVisible ) {
+		return 'DOMAINS';
 	}
-`;
+	return 'DEFAULT';
+}
+
+function renderContactDetails( {
+	translate,
+	domainName,
+	isDomainFieldsVisible,
+	contactInfo,
+	renderDomainContactFields,
+	setters,
+	PhoneInput,
+	CountrySelectMenu,
+	countriesList,
+} ) {
+	const format = contactDetailsFormat( { isDomainFieldsVisible } );
+	const requiresVatId = isEligibleForVat( contactInfo.countryCode.value );
+	switch ( format ) {
+		case 'DOMAINS':
+			return (
+				<React.Fragment>
+					<DomainContactFieldsDescription>
+						{ translate(
+							'Registering a domain name requires valid contact information. Privacy Protection is included for all eligible domains to protect your personal information.'
+						) }
+					</DomainContactFieldsDescription>
+					{ renderDomainContactFields(
+						[ domainName ],
+						prepareDomainContactDetails( contactInfo ),
+						setters.updateContactDetails,
+						setters.applyDomainContactValidationResults
+					) }
+					{ requiresVatId && <VatIdField /> }
```

Comment:
I'm surprised the VAT id isn't covered by the domain contact fields. 🤔

Reply:
I just checked again and it appears VAT is not included in the domain contact fields. I think that component is designed to collect the data needed by the registrar (it's used to let customers update their WHOIS record for instance) but VAT is only relevant for payments.